blottertrax/helper/title_parser.py

```diff
@@ -1,59 +1,137 @@
-            if feature not in post_title:
-                continue
-
-            feat_index = post_title.index(feature)
-
-            # remove featuring from artist if exists
-            if len(artist) > feat_index:
-                artist = artist[:feat_index].strip()
-
-            feat_index += len(feature)
-
-            # isolate featuring artist
-            feature_artist = post_title[feat_index:].strip()
-
-            for end_char in [' -', ')', '[', ' —']:
-                if end_char in feature_artist:
-                    split_result = feature_artist.split(end_char)
-                    feature_artist, song_title = split_result
-
-                    break
-
-        if song_title is None:
-            raise LookupError
-
-        # Remove everything between () and []
-        song_title = re.sub(r"\((.*?)\)", "", song_title)
-        song_title = re.sub(r"\[(.*?)\]", "", song_title)
-        song_title = song_title.strip()
-
-        if artist is not None:
-            return ParsedSubmission(artist, feature_artist, song_title, submission.url)
-
-        raise LookupError
+from blottertrax.value_objects.parsed_submission import ParsedSubmission
+
+class TitleParser:
+
+    # Calls the main parsing methods and returns a parsed submission if successful.
+    @classmethod
+    def create_parsed_submission_from_submission(self, submission) -> ParsedSubmission:
+        artist = self.get_artist_name_from_submission_title(submission.title)
+        song = self.get_song_name_from_submission_title(submission.title, artist)
+        return ParsedSubmission(artist[0], artist[1], song, submission.url) #, submission.id)
+
+    # If featuring artist exists, returns featuring artist, else returns main artist.
+    @staticmethod
+    def get_prioritized_artist(parsed_submission):
+        if parsed_submission.featuring_artist is None:
+            return parsed_submission.artist
+
+        return parsed_submission.featuring_artist
+
+    #iterate and remove prioritized dash. Second variable denotes whether looking for artist (0) or song (1)
+    @staticmethod
+    def remove_prioritized_dash(post_title, is_song):
+        dash_char = ['-', '—', '–']
+        # iterate each of the set dash characters
+        for dash in dash_char:
+            # First check for instances of the dash character appearing twice in a row. We want to prioritize this
+            if (dash + dash) in post_title:
+                parsed = post_title.split(dash + dash, 1)[is_song].strip() # splits off the double dash and strips
+                # Will return "a-B (FEAT c) -" for artist  and "- d [e/f](g)" for song
+                return parsed
+        
+        # if no instance of dash character appearing twice in a row just split first dash instance.
+        for dash in dash_char:
+            if dash in post_title:
+                parsed = post_title.split(dash, 1)[is_song].strip()
+                return parsed
+        
+        raise LookupError
```

Comment:
I've merged my PR before yours so could you edit this so it doesn't throw `LookupError`, instead it should return empty `ParsedResult` with the first parameter `false`. This is an indicator that the parsing didn't succeed. This was introduced so that Youtube and Souncloud could work even if the parsing of the submission title failed.

Reply:
Due to this being a subroutine that didn't have access to submission.url I ended up wrapping the parsing in create_parsed_submission_from_submission with a try catch to return the ParsedSubmission object instead.

Fixed this myself as Tc said he'd be out for a bit.